tests/php/Extension/FluentFilteredExtensionTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace TractorCow\Fluent\Tests\Extension;
+
+use Page;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Dev\SapphireTest;
+use TractorCow\Fluent\Extension\FluentFilteredExtension;
+use TractorCow\Fluent\Extension\FluentSiteTreeExtension;
+use TractorCow\Fluent\Model\Domain;
+use TractorCow\Fluent\Model\Locale;
+use TractorCow\Fluent\State\FluentState;
+
+class FluentFilteredExtensionTest extends SapphireTest
+{
+    protected static $fixture_file = 'FluentFilteredExtensionTest.yml';
+
+    protected static $required_extensions = [
+        SiteTree::class => [
+            FluentSiteTreeExtension::class,
+            FluentFilteredExtension::class,
+        ],
+    ];
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        // Clear cache
+        Locale::clearCached();
+        Domain::clearCached();
+        FluentState::singleton()
+            ->setLocale('en_NZ')
+            ->setIsDomainMode(false);
+    }
+
+    public function testAugmentSQLFrontend()
+    {
+        FluentState::singleton()
+            ->setLocale('en_NZ')
+            ->setIsFrontend(true);
+
+        $this->assertEquals(1, SiteTree::get()->count());
```

Comment:
@robbieaverill @tractorcow sorry to bug you both with this, but I'm really struggling with this test. It passes 100% of the time when I run it locally, but is failing in CI:
https://travis-ci.org/tractorcow/silverstripe-fluent/jobs/301652000

The (raw) SQL that gets added as an Inner Join in `augmentSQL` as part of this test is:
```sql
"SiteTree_FilteredLocales"."SiteTreeID" = "SiteTree"."ID" AND "SiteTree_FilteredLocales"."Fluent_LocaleID" = 1
```
Which, I think should match up fine with the fixture file that I've created? And, again, it passes locally - so, I'm really not sure where I'm going wrong.

Thank you for your help :)

Reply:
It could be due to the fixture file not being setup correctly; Please see my comment on the yml file.

Reply:
These are working now (using `FixtureFactory` rather than `yml`), however, I might still be doing wrong as between tests, the number of DB objects continues to increase (it looks like `SiteTree` objects aren't being removed).

I've got `$factory->clear()` running, but that didn't do the trick. I ended up defining an empty `yml` fixture which looked to clear the objects correctly.

I was following the docs on Fixtures (https://docs.silverstripe.org/en/4/developer_guides/testing/how_tos/fixturefactories/), but it looked outdated.

I'll be in the office tomorrow if it's easier to discuss in person.

Reply:
Well it seems green now!